internal/controller/bindings/boundendpoint_poller.go

```diff
@@ -216,22 +216,23 @@ func (r *BoundEndpointPoller) reconcileBoundEndpointsFromAPI(ctx context.Context
 	toCreate, toUpdate, toDelete := r.filterBoundEndpointActions(ctx, existingBoundEndpoints, desiredBoundEndpoints)
 
 	// create context + errgroup for managing/closing the future goroutine in the reconcile actions loops
-	errGroup, ctx := errgroup.WithContext(ctx)
-	ctx, cancel := context.WithCancel(ctx)
+	errGroup, reconcileActionCtx := errgroup.WithContext(context.Background())
+	reconcileActionCtx, cancel := context.WithCancel(reconcileActionCtx)
+	reconcileActionCtx = ctrl.LoggerInto(reconcileActionCtx, log)
 	r.reconcilingCancel = cancel
 
 	// launch goroutines to reconcile the BoundEndpoints' actions in the background until the next polling loop
 
-	r.reconcileBoundEndpointAction(ctx, errGroup, toCreate, "create", func(ctx context.Context, binding bindingsv1alpha1.BoundEndpoint) error {
-		return r.createBinding(ctx, binding)
+	r.reconcileBoundEndpointAction(reconcileActionCtx, errGroup, toCreate, "create", func(reconcileActionCtx context.Context, binding bindingsv1alpha1.BoundEndpoint) error {
+		return r.createBinding(reconcileActionCtx, binding)
 	})
 
-	r.reconcileBoundEndpointAction(ctx, errGroup, toUpdate, "update", func(ctx context.Context, binding bindingsv1alpha1.BoundEndpoint) error {
-		return r.updateBinding(ctx, binding)
+	r.reconcileBoundEndpointAction(reconcileActionCtx, errGroup, toUpdate, "update", func(reconcileActionCtx context.Context, binding bindingsv1alpha1.BoundEndpoint) error {
+		return r.updateBinding(reconcileActionCtx, binding)
 	})
 
-	r.reconcileBoundEndpointAction(ctx, errGroup, toDelete, "delete", func(ctx context.Context, binding bindingsv1alpha1.BoundEndpoint) error {
-		return r.deleteBinding(ctx, binding)
+	r.reconcileBoundEndpointAction(reconcileActionCtx, errGroup, toDelete, "delete", func(reconcileActionCtx context.Context, binding bindingsv1alpha1.BoundEndpoint) error {
+		return r.deleteBinding(reconcileActionCtx, binding)
 	})
 
 	return nil
```

Comment:
**Question** Do we need an `errGroup.Wait()` here or are we just using the errgroup to spawn all of them at the same time and relying on the context being canceled? If the latter, it might be clearer if we drop the errgroup and just use goroutines for  concurrent processing.

Reply:
No I think you're right - I'm mixing up using errGroup and managing the ctx cancel myself. I'll clean this up.